neurodamus/node.py

```diff
@@ -1421,21 +1472,20 @@ def _sim_corenrn_configure_datadir(self, corenrn_restore, coreneuron_direct_mode
                     (shm_avail >> 20),
                     (mem_avail >> 20),
                 )
-        return corenrn_datadir if not self._shm_enabled else corenrn_datadir_shm
+        SimConfig.coreneuron_datadir = (
+            corenrn_datadir if not self._shm_enabled else corenrn_datadir_shm
+        )
```

Comment:
I don't know how, this PR crashes with `--enable-shm=ON`.
But enable shm is tricky to test. First, a linux system is required, second an env variable `SHMDIR=/dev/shm` is needed.  I don't know if we have `SHMDIR` always created on bb5 in the past, but I need to create manually in my linux machine to be able to run.
I have created a new unit test https://github.com/openbraininstitute/neurodamus/pull/204 to test the shm. Feel free to include my test in your PR.

Reply:
solved and I added your PR in mine